codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/DocumentShapeSerVisitor.java

```diff
@@ -323,7 +324,12 @@ private void generateSerFunction(
         GoWriter writer = context.getWriter();
 
         Symbol symbol = symbolProvider.toSymbol(shape);
-        String functionName = ProtocolGenerator.getDocumentSerializerFunctionName(shape, context.getProtocolName());
+
+        String functionName = shape.hasTrait(SyntheticClone.class)
+                ? ProtocolGenerator.getOperationDocumentSerFuncName(
+                shape, context.getProtocolName())
+                : ProtocolGenerator.getDocumentSerializerFunctionName(
+                shape, context.getService(), context.getProtocolName());
```

Comment:
If a ternary operator spans more than one or two lines then it should just be an if/else for clarity.